plugins/@grouparoo/onesignal/src/lib/export/exportProfile.ts

```diff
@@ -34,6 +34,11 @@ export const exportProfile: ExportProfilePluginMethod = async ({
   const allKeys = new Set([...newKeys, ...oldKeys]);
 
   if (toDelete) {
+    if (!syncOperations.delete) {
+      throw new Errors.InfoError(
+        "Destination sync mode does not delete existing device tags."
+      );
+    }
     // When deleting, we just clear the old tags. No real deletion occurs.
```

Comment:
We don't need to do this anymore. Core should never send a `toDelete: true`. We were always doing this before because OneSignal only supports enriching. I think we can always throw the error if `toDelete: true`.

```js

if (toDelete) {
  throw new Errors.InfoError(
    "OneSignal does not delete existing devices."
  );
}
```

(or smething like this)

The corresponding tests for this should also be removed, and just check that it throws the error if `toDelete: true`

Reply:
ohhh I see